code/datums/brain_damage/psychological.dm

```diff
@@ -0,0 +1,301 @@
+
+
+/datum/brain_trauma/psychological/social_anxiety //ported from quirks. The quirk gives you this at round start now
+	name = "Social Anxiety"
+	desc = "Talking to people is very difficult for the subject ,often stuttering or even locking up."
+	gain_text = "<span class='danger'>You start worrying about what you're saying.</span>"
+	lose_text = "<span class='notice'>You feel easier about talking again.</span>" //if only it were that easy!
+	scan_desc = "Electric imaging suggests a mental disorder is present"
+	var/dumb_thing = TRUE
+	cure_points = 500
+	lithiated_mod = -2
+	control_mod = 2
+	fearless_mod = 1
+	relaxed_mod = 4
+
+/datum/brain_trauma/psychological/social_anxiety/on_life()
+	..()
+	if(HAS_TRAIT(owner, TRAIT_FEARLESS))
+		return
+	var/nearby_people = 0
+	for(var/mob/living/carbon/human/H in oview(3, owner))
+		if(H.client)
+			nearby_people++
+	var/mob/living/carbon/human/H = owner
+	if(prob(2 + nearby_people))
+		H.stuttering = max(3, H.stuttering)
+	else if(prob(min(3, nearby_people)) && !H.silent)
+		to_chat(H, "<span class='danger'>You retreat into yourself. You <i>really</i> don't feel up to talking.</span>")
+		H.silent = max(10, H.silent)
+	else if(prob(0.5) && dumb_thing)
+		to_chat(H, "<span class='userdanger'>You think of a dumb thing you said a long time ago and scream internally.</span>")
+		dumb_thing = FALSE //only once per life
+		if(prob(1))
+			new/obj/item/reagent_containers/food/snacks/spaghetti/pastatomato(get_turf(H)) //now that's what I call spaghetti code
+
+
+/datum/brain_trauma/psychological/social_anxiety/on_gain()
+	ADD_TRAIT(owner, TRAIT_SOCIAL_ANXIETY, PSYCH_TRAIT)
+	..()
+
+/datum/brain_trauma/psychological/social_anxiety/on_lose()
+	REMOVE_TRAIT(owner, TRAIT_SOCIAL_ANXIETY, PSYCH_TRAIT)
+	..()
+
+/datum/brain_trauma/psychological/bipolar
+	name = "Bipolar Disorder"
+	desc = "Subject experiences alternating mood swings"
+	gain_text = "<span class='danger'>You feel as on a moodswing.</span>"
+	lose_text = "<span class='notice'>You feel peace</span>" //if only it were that easy!
+	scan_desc = "Electric imaging suggests a mental disorder is present"
+	cure_points = 750
+	lithiated_mod = 2
+	control_mod = 4
+	fearless_mod = -2
+	relaxed_mod = -5
+	var/mood_swing_state = FALSE
+
+/datum/brain_trauma/psychological/bipolar/proc/change_mood_swing()
+	mood_swing_state = !mood_swing_state
+	if(mood_swing_state)
+		to_chat(owner, "<span class='danger'>Your heart begins to race. You are overfilled with joy!.</span>")
+		SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "bipolar", /datum/mood_event/mood_swing_up)
+	else
+		to_chat(owner, "<span class='danger'>You feel the stress catching up to you. Sadness fills your heart!</span>")
+		SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "bipolar", /datum/mood_event/mood_swing_down)
+	addtimer(CALLBACK(src, .proc/change_mood_swing), 2 MINUTES)
+
+/datum/brain_trauma/psychological/bipolar/on_gain()
+	change_mood_swing()
+	ADD_TRAIT(owner, TRAIT_BIPOLAR, PSYCH_TRAIT)
+	..()
```

Comment:
You should default to calling parent first in procs unless you have good reason otherwise
`. = ..()`
If you need to have it last then do the following instead
`return ..()`
Yes even if you have no return value, this is useful for later when a return value is potentially added and it's good to have standardized parent calls.